Applications/WebPortal/WebApp/library/source/components/single_security_input.tsx

```diff
@@ -0,0 +1,690 @@
+import { css, StyleSheet } from 'aphrodite';
+import * as Nexus from 'nexus';
+import * as React from 'react';
+import { Button, DisplaySize, HLine, Modal } from '..';
+
+interface Properties {
+
+  /** The size at which the component should be displayed at. */
+  displaySize: DisplaySize;
+
+  /** Determines if the component is readonly. */
+  readonly?: boolean;
+
+  /** The list of securities to display. */
+  value?: Nexus.Security;
+
+  /** Called when the list of values changes.
+   * @param value - The new security
+   */
+  onChange?: (value: Nexus.Security) => void;
+}
+
+interface State {
+  inputString: string;
+  isEditing: boolean;
+  localValue: Nexus.Security;
+  isSelected: boolean;
+}
+
+/** A component that displays a list of securities. */
+export class SingleSecurityInput extends React.Component<Properties, State> {
+  public static readonly defaultProps = {
+    value: '',
```

Comment:
```value``` is of type ```Security```.

Reply:
Is there a Nexus.Security.None type variable I can use? Or should I make it null?

Reply:
No, don't make it ```null```. Instead add a static value called ```NONE``` to Nexus/WebApi/source/definitions/security.ts, the symbol should be ```''```, the market is ```MarketCode.NONE``` and the country is ```CountryCode.NONE```.